### PR TITLE
#23: Upgrade dependencies

### LINKED ISSUE #23: Fix dependabot warning
See https://github.com/exasol/exasol-driver-ts/security/dependabot
* https://github.com/exasol/exasol-driver-ts/security/dependabot/3: tough-cookie Prototype Pollution vulnerability

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #23

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -22,13 +22,14 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node: [14, 16, 18]
+        node: ['16', '18', '20']
```
Comment (tkilias) [inline .github/workflows/pr.yml:25]: Good one, this is annoying with yaml

File: doc/developer_guide/developer_guide.md [hunk 2]
```diff
@@ -15,17 +15,17 @@ npm run lint:ci // No autofix enabled
 ### Unit tests
 
 ```bash
- npm run test // Runs both test in parallel
- npm run test:dom
- npm run test:node
+npm run test // Runs both test in parallel
```
Comment (tkilias) [inline doc/developer_guide/developer_guide.md:18]:
```suggestion
npm run test # Runs both test in parallel
```

// is not a comment in bash, which would prevent the coping of this code block

File: doc/developer_guide/developer_guide.md [hunk 3]
```diff
@@ -15,17 +15,17 @@ npm run lint:ci // No autofix enabled
...
+npm run test:node
 ```
 
 ### Integration tests
 
 ```bash
- npm run itest // Runs both test in parallel
- npm run itest:dom
- npm run itest:node
+npm run itest // Runs both test in parallel
```
Comment (tkilias) [inline doc/developer_guide/developer_guide.md:26]:
```suggestion
npm run itest # Runs both test in parallel
```

File: jest.config.ts [hunk 4]
```diff
@@ -77,7 +77,7 @@ const config: Config = {
   transformIgnorePatterns: ['node_modules/(?!.*.mjs$)'],
   preset: './jest.preset.js',
   moduleFileExtensions: ['ts', 'js', 'html'],
-  coverageDirectory: './coverage/exasol-driver',
+  coverageDirectory: './coverage',
```
Comment (tkilias) [inline jest.config.ts:80]: is there a reason for the change?

File: package.json [hunk 5]
```diff
@@ -1,48 +1,48 @@
...
-  "version": "0.1.1",
+  "version": "0.1.2",
   "description": "Exasol SQL Driver",
   "scripts": {
     "build": "rollup -c",
     "watch": "rollup -cw",
-    "test": "jest --selectProjects unit-dom unit-node",
-    "test:dom": "jest --selectProjects unit-dom",
-    "test:node": "jest --selectProjects unit-node",
+    "test": "jest --selectProjects unit-dom unit-node --coverage",
```
Comment (tkilias) [inline package.json:8]: so before you didn't run with coverage?

**Comment by tkilias** [comment]
My suggestion would be to mention some of the tweaks you did also in the final commit message, because it is not only a dependency update, it looks like.

File: doc/developer_guide/developer_guide.md (hunk 2 above)
Comment (kaklakariada) [reply doc/developer_guide/developer_guide.md:18]: Smart!

File: jest.config.ts (hunk 4 above)
Comment (kaklakariada) [reply jest.config.ts:80]: Yes, there's no need for the sub-dir. `./coverage/` is empty anyway.
Coverage was not generated before at all.

File: package.json (hunk 5 above)
Comment (kaklakariada) [reply package.json:8]: No, we didn't 😆

**Pull request merged**